fluentd/fluent-plugin-grafana-loki/lib/fluent/plugin/out_loki.rb

```diff
@@ -131,29 +134,16 @@ def write(chunk)
         body = { 'streams' => payload }
 
         # add ingest path to loki url
+        res = loki_http_request(body)
 
-        req = Net::HTTP::Post.new(
-          @uri.request_uri
-        )
-        req.add_field('Content-Type', 'application/json')
-        req.add_field('X-Scope-OrgID', @tenant) if @tenant
-        req.body = Yajl.dump(body)
-        req.basic_auth(@username, @password) if @username
-
-        opts = ssl_opts(@uri)
+        return if res.is_a?(Net::HTTPSuccess)
 
-        log.debug "sending #{req.body.length} bytes to loki"
-        res = Net::HTTP.start(@uri.host, @uri.port, **opts) { |http| http.request(req) }
-        unless res&.is_a?(Net::HTTPSuccess)
-          res_summary = if res
-                          "#{res.code} #{res.message} #{res.body}"
-                        else
-                          'res=nil'
-                        end
-          log.warn "failed to #{req.method} #{@uri} (#{res_summary})"
-          log.debug Yajl.dump(body)
+        res_summary = "#{res.code} #{res.message} #{res.body}"
+        log.warn "failed to write post to #{@uri} (#{res_summary})"
+        log.debug Yajl.dump(body)
 
-        end
+        # Only retry 429 and 500s
+        raise(LogPostError, res_summary) if res.is_a?(Net::HTTPTooManyRequests) || res.is_a?(Net::HTTPServerError)
```

Comment:
in your test it seems to also cover all 5xx right ? Like 505 will also raise an error ? Sorry I'm not familiar with what `HTTPServerError ` is.

Reply:
Yup. HTTPServerError will cover all 5xx